### PR TITLE
Add aws_pca to the spire-server

### PR DESCRIPTION
**Pull request by PetrMc**
This change allows aws_pca to be configured via values of this chart.

__Requires 1.7.1 version__ per [bug](https://github.com/spiffe/spire/issues/4351) - this will not work until 1.7.1 is released.

### CODE REVIEW

**Review by marcofranssen — APPROVED** [review]
LGTM

**Review by kfox1111 — COMMENTED** [review]
Thank you  so much for working on this.

A few formatting changes to follow convention are suggested inline.

File: charts/spire/charts/spire-server/values.yaml [hunk 1]
```diff
@@ -184,6 +184,25 @@ upstreamAuthority:
         certificate: ""
         key: ""
         bundle: ""
+  awspca:
+    enabled: false
+    # -- AWS Region to use
+    region: ""
+    # -- ARN of the "upstream" CA certificate
+    certificate_authority_arn: ""
```
[the remark this answers is not in the document]
Comment (PetrMc) [reply charts/spire/charts/spire-server/values.yaml]: applied camelCase. Thanks @kfox1111  - please review!

File: charts/spire/charts/spire-server/values.yaml [hunk 2]
```diff
@@ -170,7 +170,6 @@ ca_subject:
   country: NL
   organization: Example
   common_name: example.org
```
Comment (kfox1111) [inline charts/spire/charts/spire-server/values.yaml:172]: Blank line removed by mistake. suggestion to add it back.
```suggestion
  common_name: example.org
  
```

File: charts/spire/charts/spire-server/values.yaml [hunk 3]
```diff
@@ -184,6 +183,25 @@ upstreamAuthority:
...
         key: ""
         bundle: ""
+  awsPCA:
+    enabled: true
+    # -- AWS Region to use
+    region: "us-west-2"
+    # -- ARN of the "upstream" CA certificate
+    certificateAuthorityArn: "arn:aws:acm-pca:us-west-2:945673678728:certificate-authority/636dbcd8-203f-43e0-9701-be854b094a2b"
+    # -- (Optional) ARN of an IAM role to assume
+    assumeRoleArn: "arn:aws:iam::945673678728:role/PCAAccessSpireRole"
```
Comment (kfox1111) [inline charts/spire/charts/spire-server/values.yaml:193]: Guessing you probably don't want to be pointing everyone at your own things by default. Leaked in by mistake during testing?

File: charts/spire/charts/spire-server/values.yaml [hunk 4]
```diff
@@ -184,6 +183,25 @@ upstreamAuthority:
         certificate: ""
         key: ""
         bundle: ""
+  awsPCA:
+    enabled: true
```
Comment (kfox1111) [inline charts/spire/charts/spire-server/values.yaml:187]: Should default to off
```suggestion
    enabled: false
```

File: charts/spire/charts/spire-server/values.yaml [hunk 5]
```diff
@@ -184,6 +183,25 @@ upstreamAuthority:
...
+    # -- AWS Region to use
+    region: "us-west-2"
+    # -- ARN of the "upstream" CA certificate
+    certificateAuthorityArn: "arn:aws:acm-pca:us-west-2:945673678728:certificate-authority/636dbcd8-203f-43e0-9701-be854b094a2b"
```
Comment (kfox1111) [inline charts/spire/charts/spire-server/values.yaml:191]: ARN instead of Arn please, since its an acronym. Same throughout.

**Review by kfox1111 — COMMENTED** [review]
Thanks for the update. Some comments inline.

File: charts/spire/charts/spire-server/values.yaml (hunk 5 above)
Comment (PetrMc) [reply charts/spire/charts/spire-server/values.yaml]: addressed in the new commit

File: charts/spire/charts/spire-server/values.yaml (hunk 3 above)
Comment (PetrMc) [reply charts/spire/charts/spire-server/values.yaml]: my test file got in a way - appologies.

**Review by kfox1111 — APPROVED** [review]
LGTM

**Pull request merged**